lib/hem/tasks/magento2/createpatch.rb

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env ruby
+# ^ Syntax hint
+
+desc 'Create magento2 patch'
+task :createpatch do
+  package_name = Hem.ui.ask("which package did you modify? (e.g. module-catalog)")
+  patch_name = Hem.ui.ask("what should be the name of the patch? (e.g. my awesome fix )")
+  patch_tmp_path = '/tmp/patchtmp/'
+  package_path = 'vendor/magento/' + package_name
+
+  if patch_name.empty?
+    Kernel.abort('patch name required')
+  end
+
+  if package_name.empty?
+    Kernel.abort('package name required')
+  end
+
+  if !File.exists?(package_path)
+    Kernel.abort(package_path + ' does not exist')
+  end
+
+  Hem.ui.info('Patch will be created for ' + package_path)
+  Hem.ui.info('1. Save your changes to ' + patch_tmp_path + package_name)
+  FileUtils.rm_rf(patch_tmp_path)
+  FileUtils.mkdir_p(patch_tmp_path)
+  FileUtils.cp_r(package_path, patch_tmp_path)
+
+  Hem.ui.info('2. Reinstall package')
+  FileUtils.rm_rf(package_path)
+  run_command('php bin/composer.phar install', realtime: true, indent: 2)
```

Comment:
This can fail if the host PHP install is not set up with the right extensions or config.
`Rake::Task['deps:composer'].execute` should handle that case ( see the source here: https://github.com/inviqa/hem/blob/431174d640b025ee965602d94303c113fabf976b/lib/hem/tasks/deps.rb#L26 )

In the case of rsync handling the vendor directory, we'll need to call this: https://github.com/inviqa/hem-tasks-rsync/blob/v3.0.1/lib/hem/tasks/rsync/deps.rb#L89 to ensure the vendor directory is synced to the VM before it runs composer in the VM.
This can be done with:
`Rake::Task['deps:sync:vendor_directory_to_guest'].execute if Rake::Task.task_defined?('deps:sync:vendor_directory_to_guest')`